app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -36,16 +35,22 @@ def destroy
 
     def authenticate
       if sanitized_shop_name.present?
-        fullpage_redirect_to "#{main_app.root_path}auth/shopify?shop=#{sanitized_shop_name}"
+        session['shopify.omniauth_params'] = { shop: sanitized_shop_name }
```

Comment:
I don't think we can set the session from the child-frame until the parent frame redirects to our app. If I remember correctly, this was an issue in Safari. Safari is more strict about third party cookies by default.

Reply:
Sounds like we need to 🎩  with safari.

Reply:
This was Safari 11 I believe (and some version of IE)

Reply:
I was able to install my app using this branch using Safari on my laptop. I check the version and it is `Version 10.1.2 (12603.3.8)` - is my safari out of date?

Reply:
Apple's download link for Safari 11 is broken. @Hammadk suggested I try changing this setting:

![image](https://user-images.githubusercontent.com/1965489/29793567-4b6d8792-8c12-11e7-80be-1d8747971fc9.png)

to `Allow from current website only`

I was able to install the app fine. I also cleared all the cookies in Safari and tried again which was also successful. I think this change is okay.